component/azstorage/block_blob.go

```diff
@@ -606,10 +607,35 @@ func (bb *BlockBlob) WriteFromFile(name string, metadata map[string]string, fi *
 	//defer exectime.StatTimeCurrentBlock("WriteFromFile::WriteFromFile")()
 
 	blobURL := bb.Container.NewBlockBlobURL(filepath.Join(bb.Config.prefixPath, name))
-
 	defer log.TimeTrack(time.Now(), "BlockBlob::WriteFromFile", name)
+	blockSize := bb.Config.blockSize
+	// if the block size is not set then we configure it based on file size
+	if !config.IsSet(compName + ".block-size-mb") {
```

Comment:
We could have set the value of blockSize field in config to 0 and just checked that instead of checking IsSet in each upload call. That will be more efficient.

Reply:
if we do that then for all other calls our default block size would be 0. And in that case, we would be relying on the GO sdk logic to determine the block size for us. In our meeting we discussed we don't want to do that. Hence adding this logic to the CopyFromFile ourselves and leaving the rest as 16MB default. 
I don't think its that inefficient given that this method is `WriteFromFile` which uploads the entire file so I think its ok.

Reply:
I did not mean that. I am just suggesting instead of checking config.IsSet, the variable is already read into a field in config we can check that. It will be more efficient to check variable instead of searching into config map on every upload.

Reply:
Also, when we are initializing the class object, we are setting it to 16MB remove that and set it to 0, as there is no default block size anymore.

Reply:
ah ok yea I thought we wanted to maintain that 16MB default but if we don't care then I can just have it as 0

Reply:
done